scenes/SceneEditAccountV2.js

```diff
@@ -0,0 +1,411 @@
+  _ref;
+
+  state = {
+    username: this.props.viewer.username,
+    password: "",
+    confirm: "",
+    body: this.props.viewer.data.body,
+    photo: this.props.viewer.data.photo,
+    name: this.props.viewer.data.name,
+    deleting: false,
+    allow_filecoin_directory_listing: this.props.viewer.allow_filecoin_directory_listing,
+    allow_automatic_data_storage: this.props.viewer.allow_automatic_data_storage,
+    allow_encrypted_data_storage: this.props.viewer.allow_encrypted_data_storage,
+    changingPassword: false,
+    changingUsername: false,
+    changingAvatar: false,
+    changingFilecoin: false,
+    copyValue: "",
+    loading: false,
+    tab: 0,
+    contextMenu: null,
+    selectedIndex: -1,
+  };
+
+  componentDidMount = () => {
+    window.addEventListener("keydown", this._handleDocumentKeydown);
+    this.debounceInstance = this.debounce(() => {
+      if (this.state.selectedIndex !== -1) {
+        this.setState({ selectedIndex: -1 });
+      }
+      this.props.onSearch();
+    }, 500);
+  };
+
+  debounce = (fn, time) => {
+    let timer;
+    return () => {
+      window.clearTimeout(timer);
+      timer = window.setTimeout(() => fn(), time);
+    };
+  };
+
+  _handleUpload = async (e) => {
+    this.setState({ changingAvatar: true });
+    e.persist();
+    let file = e.target.files[0];
+
+    if (!file) {
+      dispatchCustomEvent({
+        name: "create-alert",
+        detail: {
+          alert: {
+            message: "Something went wrong with the upload. Please try again",
+          },
+        },
+      });
+      return;
+    }
+
+    // NOTE(jim): Only allow images for account avatar.
+    if (!file.type.startsWith("image/")) {
+      dispatchCustomEvent({
+        name: "create-alert",
+        detail: {
+          alert: { message: "Upload failed. Only images and gifs are allowed" },
+        },
+      });
+      return;
+    }
+
+    const response = await FileUtilities.upload({ file });
+
+    if (!response) {
+      dispatchCustomEvent({
+        name: "create-alert",
+        detail: {
+          alert: { message: "We're having trouble connecting right now" },
+        },
+      });
+      this.setState({ changingAvatar: false });
+      return;
+    }
+
+    if (response.error) {
+      dispatchCustomEvent({
+        name: "create-alert",
+        detail: { alert: { decorator: json.decorator } },
+      });
+      this.setState({ changingAvatar: false });
+      return;
+    }
+
+    const { json } = response;
+    const cid = json.data.ipfs.replace("/ipfs/", "");
+    const url = Strings.getCIDGatewayURL(cid);
+    await Actions.updateViewer({
+      data: {
+        photo: Strings.getCIDGatewayURL(cid),
+        body: this.state.body,
+        name: this.state.name,
+      },
+    });
+
+    await this.props.onRehydrate();
+    this.setState({ changingAvatar: false, photo: url });
+  };
+
+  _handleSaveAll = async (e) => {
+    this.setState({ changingUsername: true });
```

Comment:
Remove all of the other this.setStates and combine into one. This will cause problems.